lightning/src/ln/channel.rs

```diff
@@ -2578,8 +2593,11 @@ impl<Signer: Sign> Channel<Signer> {
 			if update_add_htlcs.is_empty() && update_fulfill_htlcs.is_empty() && update_fail_htlcs.is_empty() && self.holding_cell_update_fee.is_none() {
 				return Ok((None, htlcs_to_fail));
 			}
-			let update_fee = if let Some(feerate) = self.holding_cell_update_fee {
-				self.pending_update_fee = self.holding_cell_update_fee.take();
+			let update_fee = if let Some(feerate) = self.holding_cell_update_fee.take() {
+				assert!(self.is_outbound());
+				self.pending_update_fee = Some((feerate,
+					// Note that as we're an outbound channel, this is just a dummy value
+					InboundFeeUpdateState::RemoteAnnounced));
```

Comment:
Maybe a `NotApplicable` variant?

Reply:
Added an `Outbound` variant and renamed in #985 as Val suggested there.